enumerables.rb

```diff
@@ -0,0 +1,153 @@
+module Enumerable
+  def my_each
+    return to_enum :name unless block_given?
+
+    for element in self
+      yield element
+    end
+  end
+
+  def my_each_with_index
+    return to_enum :name unless block_given?
+
+    c = 0
+    for element in self
+      yield element, c
+      c += 1
+    end
+  end
+
+  def my_any?(args = nil)
+    if args.nil?
+      if block_given?
+        my_each { |x| return true if yield(x) }
+      else
+        my_each { |x| return true if x }
+      end
+    elsif args.is_a?(Regexp)
+      my_each { |x| return true if x.match(args) }
+    elsif args.is_a?(Module)
+      my_each { |x| return true if x.is_a?(args) }
+    else
+      my_each { |x| return true if x == args }
+    end
+    false
+  end
+
+  def my_none?(args = nil)
+    if args.nil?
+      if block_given?
+        my_each { |x| return false if yield(x) }
+      else
+        my_each { |x| return false if x }
+      end
+    elsif args.is_a?(Regexp)
+      my_each { |x| return false if x.match(args) }
+    elsif args.is_a?(Module)
+      my_each { |x| return false if x.is_a?(args) }
+    else
+      my_each { return false if val == args }
+    end
+    true
+  end
+
+  def my_select
+    return to_emum :my_select unless block_given?
```

Comment:
At my_select we only got one error:
1. rspec ./spec/enumerables_spec.rb:128 # enumerables #my_select returns an enumerator if no block is given

Reply:
Sorry, I don't understand why that is an error if the [docs](https://ruby-doc.org/core-2.6.5/Enumerable.html#method-i-select) is saying:
`If no block is given, an Enumerator is returned instead.`
[docs](https://ruby-doc.org/core-2.6.5/Enumerable.html#method-i-select)